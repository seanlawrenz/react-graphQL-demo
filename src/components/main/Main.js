import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { QueryRenderer } from 'react-relay';
import graphql from 'babel-plugin-relay/macro';

import environment from 'constants/environment';
import WebhookList from 'components/webhook-list/WebhookList';
import WebhookLoadingSkeleton from 'components/loading-skeletons/webhooks-loading';

// import { webhookIdGetter } from 'constants/helpers/webhookIdGetter';

import './styles.css';

const MainQuery = graphql`
  query MainQuery {
    viewer {
      ...WebhookList_viewer
    }
  }
`;

class Main extends Component {
  constructor(props) {
    super(props);

    this.deleteWebhook = this.deleteWebhook.bind(this);
  }

  componentDidMount() {
    this.mounted = true;
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  deleteWebhook(webhook) {
    // const uri = webhookIdGetter(webhook);
    // this.props.dispatch(deleteExistingWebhook(webhook, uri));
    console.log(this.props, webhook);
  }

  render() {
    return (
      <div className="gutter-bottom">
        <div className="row">
          <nav className="buttonCellTop">
            <Link to="/new" className="btn btn-link tdx-react-no-text-decoration">
              <span className="fa fa-plus fa-nopad" aria-hidden="true"></span>
              New
              <span className="sr-only">Create New</span>
            </Link>
          </nav>
          <h1 style={{ margin: '0.5em' }}>Webhooks example</h1>
        </div>
        <QueryRenderer
          environment={environment}
          query={MainQuery}
          render={({ error, props }) => {
            if (error) {
              return <div>{error.message}</div>;
            } else if (props) { // eslint-disable-line
              return <WebhookList viewer={props.viewer} />;
            }
            return <WebhookLoadingSkeleton />;
          }}
        />
      </div>
    );
  }
}

export default Main;
