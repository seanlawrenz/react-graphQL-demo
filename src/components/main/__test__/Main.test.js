import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import Main from '..';

describe('Main', () => {
  it('should render the webhooks', () => {
    const main = render(<BrowserRouter><Main /></BrowserRouter>);

    main.getByText('Webhooks example');
    expect(main.queryByText).toBeTruthy();
  });
});
