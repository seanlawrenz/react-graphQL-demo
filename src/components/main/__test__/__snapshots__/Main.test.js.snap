// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
        >
          <nav
            className="buttonCellTop"
          >
            <Link
              className="btn btn-link tdx-react-no-text-decoration"
              replace={false}
              to="/new"
            >
              <span
                aria-hidden="true"
                className="fa fa-plus fa-nopad"
              />
               New
              <span
                className="sr-only"
              >
                Create New
              </span>
            </Link>
          </nav>
          <h1
            style={
              Object {
                "margin": "0.5em",
              }
            }
          >
            Webhooks example
          </h1>
        </div>,
        <ReactRelayQueryRenderer
          environment={
            RelayModernEnvironment {
              "__setNet": [Function],
              "_deferrableSelections": Set {},
              "_network": Object {
                "execute": [Function],
                "executeWithEvents": [Function],
              },
              "_publishQueue": RelayPublishQueue {
                "_appliedOptimisticUpdates": Set {},
                "_backup": Object {},
                "_handlerProvider": [Function],
                "_pendingBackupRebase": false,
                "_pendingData": Set {},
                "_pendingOptimisticUpdates": Set {},
                "_pendingUpdaters": Set {},
                "_store": RelayMarkSweepStore {
                  "_gcEnabled": true,
                  "_hasScheduledGC": false,
                  "_index": 0,
                  "_recordSource": Object {},
                  "_roots": Map {},
                  "_subscriptions": Set {},
                  "_updatedRecordIDs": Object {},
                },
              },
              "_store": RelayMarkSweepStore {
                "_gcEnabled": true,
                "_hasScheduledGC": false,
                "_index": 0,
                "_recordSource": Object {},
                "_roots": Map {},
                "_subscriptions": Set {},
                "_updatedRecordIDs": Object {},
              },
              "configName": undefined,
              "unstable_internal": Object {
                "areEqualSelectors": [Function],
                "createFragmentSpecResolver": [Function],
                "createOperationSelector": [Function],
                "getDataIDsFromObject": [Function],
                "getFragment": [Function],
                "getRequest": [Function],
                "getSelector": [Function],
                "getSelectorList": [Function],
                "getSelectorsFromObject": [Function],
                "getVariablesFromObject": [Function],
              },
            }
          }
          query={[Function]}
          render={[Function]}
        />,
      ],
      "className": "gutter-bottom",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <nav
              className="buttonCellTop"
            >
              <Link
                className="btn btn-link tdx-react-no-text-decoration"
                replace={false}
                to="/new"
              >
                <span
                  aria-hidden="true"
                  className="fa fa-plus fa-nopad"
                />
                 New
                <span
                  className="sr-only"
                >
                  Create New
                </span>
              </Link>
            </nav>,
            <h1
              style={
                Object {
                  "margin": "0.5em",
                }
              }
            >
              Webhooks example
            </h1>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="btn btn-link tdx-react-no-text-decoration"
                replace={false}
                to="/new"
              >
                <span
                  aria-hidden="true"
                  className="fa fa-plus fa-nopad"
                />
                 New
                <span
                  className="sr-only"
                >
                  Create New
                </span>
              </Link>,
              "className": "buttonCellTop",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <span
                    aria-hidden="true"
                    className="fa fa-plus fa-nopad"
                  />,
                  " New",
                  <span
                    className="sr-only"
                  >
                    Create New
                  </span>,
                ],
                "className": "btn btn-link tdx-react-no-text-decoration",
                "replace": false,
                "to": "/new",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-plus fa-nopad",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                " New",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create New",
                    "className": "sr-only",
                  },
                  "ref": null,
                  "rendered": "Create New",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "nav",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Webhooks example",
              "style": Object {
                "margin": "0.5em",
              },
            },
            "ref": null,
            "rendered": "Webhooks example",
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "environment": RelayModernEnvironment {
            "__setNet": [Function],
            "_deferrableSelections": Set {},
            "_network": Object {
              "execute": [Function],
              "executeWithEvents": [Function],
            },
            "_publishQueue": RelayPublishQueue {
              "_appliedOptimisticUpdates": Set {},
              "_backup": Object {},
              "_handlerProvider": [Function],
              "_pendingBackupRebase": false,
              "_pendingData": Set {},
              "_pendingOptimisticUpdates": Set {},
              "_pendingUpdaters": Set {},
              "_store": RelayMarkSweepStore {
                "_gcEnabled": true,
                "_hasScheduledGC": false,
                "_index": 0,
                "_recordSource": Object {},
                "_roots": Map {},
                "_subscriptions": Set {},
                "_updatedRecordIDs": Object {},
              },
            },
            "_store": RelayMarkSweepStore {
              "_gcEnabled": true,
              "_hasScheduledGC": false,
              "_index": 0,
              "_recordSource": Object {},
              "_roots": Map {},
              "_subscriptions": Set {},
              "_updatedRecordIDs": Object {},
            },
            "configName": undefined,
            "unstable_internal": Object {
              "areEqualSelectors": [Function],
              "createFragmentSpecResolver": [Function],
              "createOperationSelector": [Function],
              "getDataIDsFromObject": [Function],
              "getFragment": [Function],
              "getRequest": [Function],
              "getSelector": [Function],
              "getSelectorList": [Function],
              "getSelectorsFromObject": [Function],
              "getVariablesFromObject": [Function],
            },
          },
          "query": [Function],
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <nav
              className="buttonCellTop"
            >
              <Link
                className="btn btn-link tdx-react-no-text-decoration"
                replace={false}
                to="/new"
              >
                <span
                  aria-hidden="true"
                  className="fa fa-plus fa-nopad"
                />
                 New
                <span
                  className="sr-only"
                >
                  Create New
                </span>
              </Link>
            </nav>
            <h1
              style={
                Object {
                  "margin": "0.5em",
                }
              }
            >
              Webhooks example
            </h1>
          </div>,
          <ReactRelayQueryRenderer
            environment={
              RelayModernEnvironment {
                "__setNet": [Function],
                "_deferrableSelections": Set {},
                "_network": Object {
                  "execute": [Function],
                  "executeWithEvents": [Function],
                },
                "_publishQueue": RelayPublishQueue {
                  "_appliedOptimisticUpdates": Set {},
                  "_backup": Object {},
                  "_handlerProvider": [Function],
                  "_pendingBackupRebase": false,
                  "_pendingData": Set {},
                  "_pendingOptimisticUpdates": Set {},
                  "_pendingUpdaters": Set {},
                  "_store": RelayMarkSweepStore {
                    "_gcEnabled": true,
                    "_hasScheduledGC": false,
                    "_index": 0,
                    "_recordSource": Object {},
                    "_roots": Map {},
                    "_subscriptions": Set {},
                    "_updatedRecordIDs": Object {},
                  },
                },
                "_store": RelayMarkSweepStore {
                  "_gcEnabled": true,
                  "_hasScheduledGC": false,
                  "_index": 0,
                  "_recordSource": Object {},
                  "_roots": Map {},
                  "_subscriptions": Set {},
                  "_updatedRecordIDs": Object {},
                },
                "configName": undefined,
                "unstable_internal": Object {
                  "areEqualSelectors": [Function],
                  "createFragmentSpecResolver": [Function],
                  "createOperationSelector": [Function],
                  "getDataIDsFromObject": [Function],
                  "getFragment": [Function],
                  "getRequest": [Function],
                  "getSelector": [Function],
                  "getSelectorList": [Function],
                  "getSelectorsFromObject": [Function],
                  "getVariablesFromObject": [Function],
                },
              }
            }
            query={[Function]}
            render={[Function]}
          />,
        ],
        "className": "gutter-bottom",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <nav
                className="buttonCellTop"
              >
                <Link
                  className="btn btn-link tdx-react-no-text-decoration"
                  replace={false}
                  to="/new"
                >
                  <span
                    aria-hidden="true"
                    className="fa fa-plus fa-nopad"
                  />
                   New
                  <span
                    className="sr-only"
                  >
                    Create New
                  </span>
                </Link>
              </nav>,
              <h1
                style={
                  Object {
                    "margin": "0.5em",
                  }
                }
              >
                Webhooks example
              </h1>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="btn btn-link tdx-react-no-text-decoration"
                  replace={false}
                  to="/new"
                >
                  <span
                    aria-hidden="true"
                    className="fa fa-plus fa-nopad"
                  />
                   New
                  <span
                    className="sr-only"
                  >
                    Create New
                  </span>
                </Link>,
                "className": "buttonCellTop",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <span
                      aria-hidden="true"
                      className="fa fa-plus fa-nopad"
                    />,
                    " New",
                    <span
                      className="sr-only"
                    >
                      Create New
                    </span>,
                  ],
                  "className": "btn btn-link tdx-react-no-text-decoration",
                  "replace": false,
                  "to": "/new",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-plus fa-nopad",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  " New",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Create New",
                      "className": "sr-only",
                    },
                    "ref": null,
                    "rendered": "Create New",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "nav",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Webhooks example",
                "style": Object {
                  "margin": "0.5em",
                },
              },
              "ref": null,
              "rendered": "Webhooks example",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "environment": RelayModernEnvironment {
              "__setNet": [Function],
              "_deferrableSelections": Set {},
              "_network": Object {
                "execute": [Function],
                "executeWithEvents": [Function],
              },
              "_publishQueue": RelayPublishQueue {
                "_appliedOptimisticUpdates": Set {},
                "_backup": Object {},
                "_handlerProvider": [Function],
                "_pendingBackupRebase": false,
                "_pendingData": Set {},
                "_pendingOptimisticUpdates": Set {},
                "_pendingUpdaters": Set {},
                "_store": RelayMarkSweepStore {
                  "_gcEnabled": true,
                  "_hasScheduledGC": false,
                  "_index": 0,
                  "_recordSource": Object {},
                  "_roots": Map {},
                  "_subscriptions": Set {},
                  "_updatedRecordIDs": Object {},
                },
              },
              "_store": RelayMarkSweepStore {
                "_gcEnabled": true,
                "_hasScheduledGC": false,
                "_index": 0,
                "_recordSource": Object {},
                "_roots": Map {},
                "_subscriptions": Set {},
                "_updatedRecordIDs": Object {},
              },
              "configName": undefined,
              "unstable_internal": Object {
                "areEqualSelectors": [Function],
                "createFragmentSpecResolver": [Function],
                "createOperationSelector": [Function],
                "getDataIDsFromObject": [Function],
                "getFragment": [Function],
                "getRequest": [Function],
                "getSelector": [Function],
                "getSelectorList": [Function],
                "getSelectorsFromObject": [Function],
                "getVariablesFromObject": [Function],
              },
            },
            "query": [Function],
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
